src/Compilers/CSharp/Portable/Binder/Semantics/Operators/UnaryOperatorEasyOut.cs

```diff
@@ -161,7 +161,7 @@ private void UnaryOperatorEasyOut(UnaryOperatorKind kind, BoundExpression operan
 
             Conversion? conversion = Conversions.FastClassifyConversion(operandType, signature.OperandType);
 
-            Debug.Assert(conversion.HasValue && conversion.Value.IsImplicit);
+            Debug.Assert(conversion is { HasValue: true, Value: { IsImplicit: true } });
```

Comment:
1. compiler could execute these in arbitrary order right?  so 'Value' might throw.
2. should we not offer if it makes the code *longer*?  Almost everything i've seen has made it smaller, but this does not.  
3.  Converting simple bool ops to patterns+bool-constants actually seems worse in some cases.

Reply:
> compiler could execute these in arbitrary order right? so 'Value' might throw.

No. The outermost `{ }` are a null check, so this has been incorrectly rewritten.

@alrz action item is to correctly handle `Nullable<T>` leveraging the pattern's null checking functionality, where in this case we should have:

```
Debug.Assert(conversion is { IsImplicit: true });
```

> should we not offer if it makes the code longer?
>
> Converting simple bool ops to patterns+bool-constants actually seems worse in some cases.

This is a refactoring, not a diagnostic. It is offered where valid, and the user can decide whether or not to apply it. There is no 'Fix All' functionality for this.